File: leetcode/455.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()

        right = len(s) - 1
        result = 0

        for i in range(len(g) - 1, -1, -1):
            if right >= 0 and g[i] <= s[right]:
                result += 1
                right -= 1
        
        return result

File: code/455.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);

        int result = 0;
        int gIndex = 0;
        int sIndex = 0;

        while (gIndex < g.length && sIndex < s.length) {
            if (s[sIndex++] >= g[gIndex]) {
                gIndex++;
                result++;
            }
        }
        return result;
    }
}